client/src/app/components/forms/inputs/forms/input-ports-form.component.ts

```diff
@@ -0,0 +1,57 @@
+import {Component, Input} from "@angular/core";
+import {CltEditorService} from "../../../clt-editor/shared/clt-editor.service";
+import {InputProperty} from "../../../../models/input-property.model";
+import {InputPortListComponent} from "../types/input-port-list.component";
+import {SidebarType} from "../../../sidebar/shared/sidebar.type";
+import {SidebarEvent, SidebarEventType} from "../../../sidebar/shared/sidebar.events";
+import {BehaviorSubject} from "rxjs";
+
+require("./form.components.scss");
+require("./input-ports-form.component.scss");
+
+@Component({
+    selector: 'inputs-ports-form',
+    directives: [InputPortListComponent],
+    template: `
+        <form>
+         <fieldset class="form-group">
+            <label>Input ports</label>
+            
+            <button type="button" class="btn btn-secondary hide-btn">Hide</button>
+
+            <input-port-list [(portList)]="inputPorts"></input-port-list>
+         
+        </fieldset>
+        <button type="button" class="btn btn-secondary add-input-btn" 
+                (click)="addInput()">Add Input</button>
+        </form>
+    `
+})
+export class InputPortsFormComponent {
+
+    @Input()
+    private inputPorts: Array<InputProperty> = [];
```

Comment:
Attributes marked as `@Input`s are by definition used from outside the component, which makes them inherently public, not private.